### DISCUSSION TITLE
Is it possible to create a new post activity by reference?

### DISCUSSION THREAD

**Discussion by markcellus** [mastodon/mastodon#18390] [Q&A] [answered]
I'm attempting to post a new message to Mastodon via ActivityPub protocol. Per "The Message" section of the [How to implement a basic ActivityPub server](https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/#the-message) docs, it shows the Create Activity with an object that contains the Note contents in the `object.content` field. 

However, I've already stored the note content on my blog at a proper URI (`id`). Is it possible to just link the new Create Activity to the post URI that is already available (maybe by omitting `object.content` and using `object.url` instead)? Or is the `object.content` required? 

Just trying to avoid having a duplicate of my post on Mastodon. :sweat_smile: 

Thanks for any information!

**Comment by trwnh** [comment, marked as answer]
>However, I've already stored the note content on my blog at a proper URI (id). Is it possible to just link the new Create Activity to the post URI that is already available

Does the URI accept the ActivityStreams profile and return a Note object in JSON-LD? If so, then yes, you can refer to it with a Create activity where `object` is just a bare IRI, I'm fairly sure

**Reply by markcellus** [reply]
Ah I see. Right now, the URI would just lead to the formatted HTML page of the blog entry. So it sounds like I'd have to make a new, separate URI that responds with the Note object in JSON-LD and just set `object` to that?

**Reply by trwnh** [reply]
Yes, but it can be the same URI if you set up your HTTP server to respond to `Accept: application/ld+json; profile="https://www.w3.org/ns/activitystreams"` properly. See https://www.w3.org/TR/activitypub/#retrieving-objects for more information

**Reply by markcellus** [reply]
Ah that's even better than what I was thinking! This is very helpful. Thank you so much :1st_place_medal:
